src/main/java/org/webjars/WebJarVersionLocator.java

```diff
@@ -47,47 +68,86 @@ public WebJarVersionLocator() {
+     *
+     * <p>See {@link WebJarVersionLocator#path(String, String)} for a detailed explanation of how the versioned file path is built.
+     *
+     * <p><b>Note:</b> This method does not perform any checks if the resulting path references an existing file.
+     *
+     * @param webJarName The name of the WebJar
+     * @param filePath   The path to the file within the WebJar
+     * @return The versioned path to the file in the classpath; null if no version for {@code webJarName} is known
+     * @see WebJarVersionLocator#path(String, String)
+     * @see WebJarVersionLocator#WEBJARS_PATH_PREFIX
      */
     @Nullable
-    public String fullPath(final String webJarName, final String exactPath) {
-        final String version = version(webJarName);
+    public String fullPath(final String webJarName, final String filePath) {
+        final String path = path(webJarName, filePath);
 
-        if (!isEmpty(version)) {
-            // todo: not sure why we check this
-            if (!exactPath.startsWith(version)) {
-                 return String.format("%s/%s/%s/%s", WEBJARS_PATH_PREFIX, webJarName, version, exactPath);
-            }
-            else {
-                return String.format("%s/%s/%s", WEBJARS_PATH_PREFIX, webJarName, exactPath);
-            }
+        if (notEmpty(path)) {
+            return String.format("%s/%s", WEBJARS_PATH_PREFIX, path);
         }
 
         return null;
     }
 
     /**
+     * Builds the versioned path for a file of a WebJar relative to the standard WebJar classpath location (see {@link WebJarVersionLocator#WEBJARS_PATH_PREFIX}).
+     *
+     * <p>The path is built by joining the {@code webJarName}, the known version and the {@code filePath}, if no version (from classpath checking) is known for the WebJar this method returns {@code null}.
+     *
+     * <p><b>Note:</b> In cases where the {@code filePath} parameter already starts with the known version of the WebJar, the version will not be added again. But it is recommended that you do NOT include a hard-coded version when looking up WebJar file paths.
      *
-     * @param webJarName The name of the WebJar, i.e. bootstrap
-     * @param exactPath The path to the file within the WebJar, i.e. js/bootstrap.js
-     * @return The path to the file in the standard WebJar classpath location, including the version, i.e. bootstrap/3.1.1/js/bootstrap.js
+     * <pre>{@code
+     * // returns "bootstrap/3.1.1/js/bootstrap.js"
+     * locator.path("bootstrap", "css/bootstrap.css");
+     *
+     * // returns "bootstrap/3.1.1/js/bootstrap.js" as well
+     * locator.path("bootstrap", "3.1.1/css/bootstrap.css");
+     *
+     * // returns null, assuming there is no "unknown" WebJar
+     * locator.path("unknown", "some/file.css");
+     * }</pre>
+     *
+     * <p><b>Note:</b> When the {@code filePath} starts with a version, that is not the correct version of the WebJar, this method returns a path with both versions.
```

Comment:
I'm not sure we need to point out this implementation detail in the Javadoc.

Reply:
You are right. I was unsure on how much details we need, so I might have added a bit too much. Its gone now.